codalab/django-s3direct
# Fix Mutation Event issues by using Mutation Observer instead

@Didayolo 

This is a fix of the Chrome issue reported here : https://github.com/codalab/codalab-competitions/issues/3503

It is based on https://github.com/bradleyg/django-s3direct/commit/d7d4d292f69c9b85a8d1e8c4cc96e4ae7e93df72 with some modification since this code alone didn't work.

To test this code on Codalab:
- Clone codalab into a new folder
- Switch branch to chromeFix
- Connect a working MinIO to codalab through the `.env`
- Build the stack `docker compose up -d`
- Open Chrome, go to localhost then upload a competition
- Make a submission on this new competition. It should work fine. The error in the Chrome dev console should also be gone 

I don't have any large submission to test so it would be nice if this can be tested too.
